### ISSUE TITLE
Disconnect at beginning of chat forced rekey; only fixed by restarting

### ISSUE THREAD

**Issue by timmc** [keybase/client#5972]
I initiated a chat with someone, turned off my computer, and when I came back the chat was borked -- they couldn't read any new messages from me. Exact sequence follows:

1. `run_keybase`
2. Follow a person
3. Initiate chat with person (send a "testing" message) -- I've never used chat before
4. Edit that message
5. Turn off computer
6. Chat partner sees message, including edit, but also sees an indicator that I need to rekey
7. I come back online and try to send a second message, and partner sees that I have sent a message, but it won't display because I still need to rekey
8. I restart the app and they receive my second message and we are able to chat

I'm using keybase `1.0.18-20170220171343+19c204b` on Debian Linux 8 (Jessie).

**Comment by timmc** [comment]
Log ID: 04cb73c5ac1ddb90d149ab1c

**Comment by maxtaco** [comment]
cc: @mmaxim can you triage this one?

**Comment by timmc** [comment]
Received some new information, edited the blow-by-blow (edits are marked, at end.)

**Comment by mmaxim** [comment]
Do you know approximatley what time you turned off the computer?

**Comment by timmc** [comment]
Yes... some time between 17:49 and 18:10 EST -- shortly after sending the first message. And I turned it on again about 20:40 EST.

**Comment by mmaxim** [comment]
Any chance your friend can log send too? Thanks for the great report!

**Comment by mantis369** [comment]
Hi. I am the other party.

Log ID: be307c788d522186b2fa7e1c

**Comment by mmaxim** [comment]
Awesome, thanks!

**Comment by mmaxim** [comment]
@strib @songgao This folder doesn't seem to get rekeyed until @timmc restart his app. Any idea why? Is this a service caching problem again?

**Comment by songgao** [comment]
This seems to be a different case. If I understand correctly, there isn't any device change (add device, revoke device, etc.) from either party throughout this time?

**Comment by mmaxim** [comment]
It looks like @mantis369 provisions at the beginning of his log.

**Comment by songgao** [comment]
Ah right

**Comment by strib** [comment]
@mmaxim: I see an issue where @timmc didn't check with the mdserver for new folders to rekey when it came back from being suspended and reestablished connectivity with the mdserver.  It probably would have taken a full hour to check in again, and then it would have rekeyed; the manual restart fixed it.  Should be an easy fix in KBFS, I'll have a PR up soon.

So that explains why the rekey didn't happen right after @timmc's device came back from being suspended, but it doesn't really explain why @mantis369 never saw the second chat message, as far as I can tell.
